### ISSUE TITLE
cy.click() doesn't trigger click event if el to click has "tabindex=0" outside of current viewport

### ISSUE THREAD

**Issue by tho-masn** [cypress-io/cypress#8279]
Hi there,

### Current behavior:

I'm using `tabindex="0"` across a vue app to be able to close forms with the ESC key. This feature introduces the following issue: In the tests the `cy.click()` calls in some cases are not triggering the actual event. Instead it seems like the click just sets the focus.

### Desired behavior:

Click event should get triggered.

### Test code to reproduce

I've tried to reproduce it with the cypress-realworld-app. You can check it out here: https://github.com/tho-masn/cypress-realworld-app/pull/1/files

To reproduce, just run the following test file: `cypress/tests/ui/bankaccounts.spec.ts`
You will see, that the click event doesn't get fired. There should appear a message below the "save" button saying "Submit worked". This issue is introduced by adding `tabIndex={0}` to the form in `BankAccountForm.tsx`

There are several options to make this example work again:

- Remove the `tabIndex={0}` attribute again
  - Not an option for me, as I need it to close the forms with ESC
- Add `{ force: true }` to the submit click call in the test.
  - Not an option for me, as I would have to add `force: true` to basically every click call
- Don't use a `div` with a click action handler for the submit button, but a real submit button
  - Not an option for me, as I have click handlers on divs which are not buttons
- Reduce the height of the div below the save button to `500`
  - This one is really weird. The issue only appears, if the page grows

### Versions

Cypress: v4.12.1
OS: MacOS 10.15.5

**Comment by jennifer-shehane** [comment]
I can recreate this with the branch provided. You can see that during the click - it's highlighting the 'Bank Account' navigation on the left, as if it's focusing there for some reason. 

![](http://g.recordit.co/uZsBq1M9m6.gif)

It'd be great to have a simpler example. I've tried to recreate outside of the RWA and haven't been successful. 

**Doesn't reproduce**

```html
<html>
<body>
  <div>
    <a tabindex="0">Link 1</a>
    <div tabindex="0"></div>
  </div>
  <form tabindex="0">
    <input/>
    <div id='save'>Save</div>
    <div id='success'></div>
  </form>
  <script>
    const saveBtn = document.getElementById('save')
    const successMsg = document.getElementById('success')

    saveBtn.addEventListener('click', (e) => {
      successMsg.innerHTML = "Success!"
    })
  </script>
</body>
</html>
```

```js
it('test', () => {
  cy.visit('index.html')
  cy.get('#save').click()
  cy.get('#success').should('contain', 'Success')
})
```

**Comment by tho-masn** [comment]
The issue is only happening, if the clicked element is out of view. I added a `div` to your code, which pushes the button out of the view:

```
<html>
<body>
  <div>
    <a>Link 1</a>
    <div></div>
  </div>
  <form tabindex="0">
    <input/>
    <div style="height: 2000px"></div>
    <div id='save'>Save</div>
    <div id='success'></div>
  </form>
  <script>
    const saveBtn = document.getElementById('save')
    const successMsg = document.getElementById('success')

    saveBtn.addEventListener('click', (e) => {
      successMsg.innerHTML = "Success!"
    })
  </script>
</body>
</html>
```

**Comment by jennifer-shehane** [comment]
Perfect! This issue does not exist in Cypress 3.4.1, so is a regression that was introduced in 3.5.0.

### Repro

```html
<html>
<body>
  <!-- The test passes is you remove tabindex here -->
  <form tabindex="0">
    <div style="height: 2000px"></div>
    <div id="save">Save</div>
    <div id="success"></div>
  </form>
  <script>
    const saveBtn = document.getElementById('save')
    const successMsg = document.getElementById('success')

    saveBtn.addEventListener('click', (e) => {
      successMsg.innerHTML = "Success!"
    })
  </script>
</body>
</html>
```

`spec.js`

```js
it('test', () => {
  cy.visit('index.html')
  cy.get('#save').click()
  cy.get('#success').should('contain', 'Success')
})
```

**Comment by sainthkh** [comment]
Interesting note. This works fine in Firefox.

**Comment by sainthkh** [comment]
This happens because of the code below:

https://github.com/cypress-io/cypress/blob/133c9706170a5b70f0ed4e19fb03b12d9d1f1fb1/packages/driver/src/cy/mouse.js#L527-L559

`mouseDownPhase.targetEl` and `mouseUpPhase.targetEl` are different. Because of that, click event isn't sent to the button. 

I couldn't verify whether it's the problem of Cypress or Chrome.

**Comment by tgreen7** [comment]
https://github.com/cypress-io/cypress/blob/133c9706170a5b70f0ed4e19fb03b12d9d1f1fb1/packages/driver/src/cy/mouse.js#L453-L484

if the code on line 482 is changed to
```focused.fireFocus($elToFocus.get(0), { preventScroll: true })```

and then those options are passed through in focused.js

```$elements.callNativeMethod(el, 'focus', opts)```

That scroll is stopped and @jennifer-shehane repro is fixed

**Comment by bahmutov** [comment]
Released in v6.7.0
